### ISSUE TITLE
Fix build warning + plugins assemble 

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#2629]
The PR #2618 that was fixing the gradle warning "Execution optimizations have been disabled for task" has been reverted (2b8556b91) because it was preventing the creation of the plugins artefacts in the project `build/` path that's required for the CI integration tests. 

When running `./gradlew assemble` the build path should result into this structure: 

```
» tree build
build
├── libs
│   └── nextflow-prj-22.01.0-edge-test-fixtures.jar
├── plugins
│   ├── nf-amazon-1.5.0
│   │   ├── META-INF
│   │   ├── classes
│   │   └── lib
│   ├── nf-amazon-1.5.0.zip
│   ├── nf-azure-0.12.0
│   │   ├── META-INF
│   │   ├── classes
│   │   └── lib
│   ├── nf-azure-0.12.0.zip
│   ├── nf-console-1.0.3
│   │   ├── META-INF
│   │   ├── classes
│   │   └── lib
│   ├── nf-console-1.0.3.zip
│   ├── nf-ga4gh-1.0.3
│   │   ├── META-INF
│   │   ├── classes
│   │   └── lib
│   ├── nf-ga4gh-1.0.3.zip
│   ├── nf-google-1.1.4
│   │   ├── META-INF
│   │   ├── classes
│   │   └── lib
│   ├── nf-google-1.1.4.zip
│   ├── nf-sqldb-0.3.0
│   │   ├── META-INF
│   │   ├── classes
│   │   └── lib
│   ├── nf-sqldb-0.3.0.zip
│   ├── nf-tower-1.4.0
│   │   ├── META-INF
│   │   ├── classes
│   │   └── lib
│   └── nf-tower-1.4.0.zip
└── tmp
    └── testFixturesJar
        └── MANIFEST.MF
```

Reverting the PR solved the build issue, but still the build produces plenty of warning. 

A second issue is that when running `gradle assemble` the plugins subprojects assemble is not called, even in the current form, I had to add an explicit `gradlew plugins:assemble` to the build. It would be nice that just the main `gradlew assemble` is enough to call it for all (sub)projects 

https://github.com/nextflow-io/nextflow/blob/d1b4cb35a4db373c188553021b66049bca7691c1/Makefile#L46-L46

**Comment by pditommaso** [comment]
Minor, instead of using gradle dynamic task names, would not make sense to use task `input` and `output` to define the dependencies as suggested here? 

https://discuss.gradle.org/t/gradle-7-0-seems-to-take-an-overzealous-approach-to-inter-task-dependencies/39656

**Comment by jorgeaguileraseqera** [comment]
about dynamic task names, if I'm not wrong, the problem is we don't have a single task because we are including them into the subproject closure and this creates the same task in every project.

**Comment by pditommaso** [comment]
But still they should have a unique name as `:plugins:foo:assemble` (where `foo` is the subproject name)

**Comment by pditommaso** [comment]
Solved by #2634
